packages/identifier-resolution/src/functions/externalOIDFIdentifier.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  ErrorMessage,
+  ExternalIdentifierOIDFEntityIdOpts,
+  ExternalIdentifierOIDFEntityIdResult,
+  ExternalJwkInfo,
+  PublicKeyHex,
+  TrustedAnchor,
+} from '../types'
+import { IAgentContext } from '@veramo/core'
+import { IOIDFClient } from '@sphereon/ssi-sdk.oidf-client'
+import { contextHasPlugin } from '@sphereon/ssi-sdk.agent-config'
+import { IJwsValidationResult } from '../types/IJwtService'
+
+/**
+ * Resolves an OIDF Entity ID against multiple trust anchors to establish trusted relationships
+ *
+ * @param opts Configuration options containing the identifier to resolve and trust anchors to validate against
+ * @param context Agent context that must include the OIDF client plugin and JWT verification capabilities
+ *
+ * @returns Promise resolving to an ExternalIdentifierOIDFEntityIdResult containing:
+ *  - trustedAnchors: Record mapping trust anchors to their public key hexes
+ *  - errorList: Optional record of errors encountered per trust anchor
+ *  - jwks: Array of JWK information from the trust chain
+ *  - trustEstablished: Boolean indicating if any trust relationships were established
+ *
+ * @throws Error if trust anchors are missing or JWT verification plugin is not enabled
+ */
+export async function resolveExternalOIDFEntityIdIdentifier(
+  opts: ExternalIdentifierOIDFEntityIdOpts,
+  context: IAgentContext<IOIDFClient>
+): Promise<ExternalIdentifierOIDFEntityIdResult> {
+  let { trustAnchors, identifier } = opts
+
+  if (!trustAnchors || trustAnchors.length === 0) {
+    return Promise.reject(Error('ExternalIdentifierOIDFEntityIdOpts is missing the trustAnchors'))
+  }
+
+  if (!contextHasPlugin(context, 'jwtVerifyJwsSignature')) {
+    return Promise.reject(Error('For OIDFEntityId resolving the agent needs to have the JwtService plugin enabled'))
+  }
+
+  const trustedAnchors: Record<TrustedAnchor, PublicKeyHex> = {}
+  const errorList: Record<TrustedAnchor, ErrorMessage> = {}
+  const jwkInfos: Array<ExternalJwkInfo> = []
+
+  for (const trustAnchor of trustAnchors) {
+    const resolveResult = await context.agent.resolveTrustChain({
+      entityIdentifier: identifier,
+      trustAnchors: [trustAnchor]
+    })
+
+    if (resolveResult.error || !resolveResult.trustChain) {
+      errorList[trustAnchor] = resolveResult.errorMessage ?? 'unspecified'
+    } else {
+      const trustChain: ReadonlyArray<string> = resolveResult.trustChain.asJsReadonlyArrayView()
+      if (trustChain.length === 0) {
+        errorList[trustAnchor] = 'Trust chain is empty'
+        continue
+      }
+      
+      const jwt = trustChain[trustChain.length - 1]
+      const jwtVerifyResult: IJwsValidationResult = await context.agent.jwtVerifyJwsSignature({ jws: jwt })
```

Comment:
Why are we verifying? The resolveTrustchain already has done that

Reply:
It's to get back indeed to load the mandatory JWK & publicKeyHex fields.
I do not verify the others anymore because the OIDF client already verified these indeed